syntax = "proto3";

package messages;

import "proto/structs.proto";

// Request for the datacenter name
message DatacenterRequest {}

message DatacenterReply {
    string datacenter = 1;
}

// Request for servers to start messaging
message BeginRequest {
    // Duration (seconds) the servers will exchange messages for
    uint32 duration = 1;
}

message BeginReply {
    ReplyCode code = 1;
}

// Request for the message log of the server
message LogRequest {}

message LogReply {
    ReplyCode code = 1;
    string address = 2;
    bool sequencer = 3;
    repeated string log = 4;
}

service Client {
    rpc datacenter(DatacenterRequest) returns (DatacenterReply) {}
    rpc begin (BeginRequest) returns (BeginReply) {}
    rpc log(LogRequest) returns (LogReply) {}
}